GeneralisedFilters/src/algorithms/particles.jl

```diff
@@ -44,14 +44,7 @@ function initialise(
         initialise_particle(rng, prior, algo, ref; kwargs...)
     end
 
-    # Set equal weights: log_w = -log(N) so weights sum to 1
-    log_weight = -log(N)
```

Comment:
Ah. I thought setting the weights to be -logN at initialisation was important for keeping the ll increments clean, but perhaps that was not the case and you've found a way to keep it looking nice.

I will look through the rest and return to this comment if there is more to say.